### PR TITLE
talk about steps more than monads in docs

### PR DESCRIPTION
**Pull request by dchelimsky**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ StateFlow is a testing framework designed to support the composition and reuse o
 ## Flows
 
 A flow is a sequence of steps to be executed with some state as a
-reference. Each step can be any of a primitive ([described below](#primitives)), a
-vector of bindings ([described below](#bindings)), or a nested flow. Flows can be
+reference. Each step can be a primitive step ([described below](#primitive-steps)), a
+vector of bindings ([described below](#bindings)), or even a nested flow. Flows can be
```
Comment (sovelten) [inline README.md]:
```suggestion
vector of bindings ([described below](#bindings)), or even another flow. Flows can be
```

Comment (sovelten) [inline README.md]: Don't want to  be too nitpicky about it, but I was thinking about the following ontology:
* A flow is a sequence of steps or bindings
* A step is a primitive step or flow
* A binding is a vector of pairs of symbols and steps (or a `:let` with a vector of regular let-bindings)

Comment (dchelimsky) [reply README.md]: Be nitpicky! I think what you say is clear. I'll update.

File: README.md [hunk 2]
```diff
@@ -4,17 +4,19 @@
...
-`def`'d to vars, and nested arbitrarily deeply.
+## Flows
 
-We define a flow with the `flow` macro:
+A flow is a sequence of steps or bindings to be executed with some state as a
+reference. Use the `flow` macro to define a flow:
 
 ```clojure
-(flow <description> <flow/bindings/primitive>*)
+(flow <description> <primitive-step/bindings/flow>*)
```
Comment (sovelten) [inline README.md]:
```suggestion
(flow <description> <step/bindings>*)
```

File: README.md [hunk 3]
```diff
@@ -158,8 +158,16 @@ out of flows.
...
 
-`match?` is a flow that will make a `clojure.test` assertion and the [`nubank/matcher-combinators`](https://github.com/nubank/matcher-combinators/) library
-for the actual checking and failure messages. `match?` asks for a string description, a value (or a flow returning a value) and a matcher-combinators matcher (or value to be checked against). Not passing a matcher defaults to `matchers/embeds` behaviour.
+`match?` produces a flow that will make a `clojure.test` assertion. It
+uses the
+[`nubank/matcher-combinators`](https://github.com/nubank/matcher-combinators/)
+library for the actual check and failure messages. `match?` asks for:
+
+* a string description
+* the actual value, or a step (or flow) which will produce it
```
Comment (sovelten) [inline README.md]: just `step`, it is understood a step can be a flow

File: doc/walkthrough.repl [hunk 4]
```diff
@@ -73,8 +84,8 @@
 ;; flows
 ;; -------------------------------------------
 ;;
-;; We use flows to string together several monads in a single
-;; monad:
+;; We use flows to string together several steps (monads) in a single
+;; step (monad):
```
Comment (sovelten) [inline doc/walkthrough.repl]: not sure if mentioning monads helps here

File: doc/walkthrough.repl [hunk 5]
```diff
@@ -120,8 +131,8 @@
 ;; programming model
 ;; -------------------------------------------
 ;;
-;; Since monads are values, we can use all the familiar tools of Clojure
-;; to define and compose them.
+;; Since steps and flows are monads, which are values, we can use all
```
Comment (sovelten) [inline doc/walkthrough.repl]:
```suggestion
;; Since steps and flows are state monads, which are values, we can use all
```

**Pull request merged**